sys/malloc.c

```diff
@@ -9,43 +9,36 @@ extern uint8_t __ebss[];
 static struct {
   void *ptr; /* Pointer to the end of kernel's bss. */
   void *end; /* Limit for the end of kernel's bss. */
-  mtx_t lock;
   bool shutdown;
-} sbrk = {__ebss, __ebss + 512 * PAGESIZE, MTX_INITIALIZER, false};
+} sbrk = {__ebss, __ebss + 512 * PAGESIZE, false};
 
 void kernel_brk(void *addr) {
   if (sbrk.shutdown)
     panic("Trying to use kernel_brk after it's been shutdown!");
-  mtx_lock(sbrk.lock);
```

Comment:
Right! The code implicitely assumes interrupts are not running. I'm not sure what to do with it. I'll just leave synchronization markers for the time being.